File: rb/features.rb
# GithubApi2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module GithubApi2Features
  def self.make_feature(name)
    case name
    when "base"
      GithubApi2BaseFeature.new
    when "test"
      GithubApi2TestFeature.new
    else
      GithubApi2BaseFeature.new
    end
  end
end

File: py/githubapi2_sdk/features.py
# GithubApi2 SDK feature factory

from githubapi2_sdk.feature.base_feature import GithubApi2BaseFeature
from githubapi2_sdk.feature.test_feature import GithubApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: GithubApi2BaseFeature(),
        "test": lambda: GithubApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
